app/src/organisms/ErrorRecoveryFlows/RecoveryDoorOpen.tsx

```diff
@@ -32,21 +33,71 @@ export function RecoveryDoorOpen({
   recoveryActionMutationUtils,
   runStatus,
   routeUpdateActions,
+  recoveryMap,
+  recoveryCommands,
 }: RecoveryContentProps): JSX.Element {
   const {
     resumeRecovery,
     isResumeRecoveryLoading,
   } = recoveryActionMutationUtils
-  const { stashedMap, proceedToRouteAndStep } = routeUpdateActions
+  const {
+    stashedMap,
+    proceedToRouteAndStep,
+    handleMotionRouting,
+  } = routeUpdateActions
+  console.log('recoveryMap: ', recoveryMap)
   const { t } = useTranslation('error_recovery')
 
+  const handleHomeAllAndRoute = (
+    route: RecoveryRoute,
+    step?: RouteStep
+  ): void => {
+    void handleMotionRouting(true, RECOVERY_MAP.ROBOT_IN_MOTION.ROUTE)
+      .then(() => recoveryCommands.homeAll())
+      .finally(() => handleMotionRouting(false))
+      .then(() => proceedToRouteAndStep(route, step))
+  }
+
   const primaryOnClick = (): void => {
-    void resumeRecovery().then(() => {
-      // See comments above for why we do this.
-      if (stashedMap != null) {
-        void proceedToRouteAndStep(stashedMap.route, stashedMap.step)
-      }
-    })
+    if (
+      recoveryMap.route === RECOVERY_MAP.MANUAL_REPLACE_STACKER_AND_RETRY.ROUTE
+    ) {
+      handleHomeAllAndRoute(
+        RECOVERY_MAP.MANUAL_REPLACE_STACKER_AND_RETRY.ROUTE,
+        RECOVERY_MAP.MANUAL_REPLACE_STACKER_AND_RETRY.STEPS.CONFIRM_RETRY
+      )
+    } else {
+      void resumeRecovery().then(() => {
+        // See comments above for why we do this.
+        if (stashedMap != null) {
+          void proceedToRouteAndStep(stashedMap.route, stashedMap.step)
+        }
+      })
+    }
+  }
+
+  const buildSubtext = (): string => {
+    if (
+      recoveryMap.route === RECOVERY_MAP.MANUAL_REPLACE_STACKER_AND_RETRY.ROUTE
+    ) {
+      return t('stacker_door_open_robot_home')
+    } else return t('close_the_robot_door')
```

Comment:
```suggestion
    } else {
    return t('close_the_robot_door')
    }
```
Always braces. It prevents occasional `return` bugs and also helps with readability!